colossus-docs/src/main/paradox/clients.md

```diff
@@ -0,0 +1,145 @@
+sent back as a failure and no subsequent action is taken.
+
+If a Connection Failure occurs (example: connection is closed, host 
+unavailable) the response is sent back as a failure and the retry policy (more
+below) takes effect.
+
+
+### Retrying Requests
+
+To enable retries, the LoadBalancingClient (LBC) should be used.  The LBC takes
+in a generator that builds a ServiceClient from an InetSocketAddress and a list
+of InetSocketAddresses.  The LBC defaults to using an unused host for a request
+or the max retry number, whichever is lower.  An example: if the retry count is
+set to four but there are only two hosts, it'll only attempt a request twice.
+
+Any failure occurring before the client receives the response (example: 
+connection closed, host not responding) is treated as an attempt and is 
+retried on the next connection.  If the maximum number of tries are exhausted, 
+a SendFailedException is returned with the last exception.
+
+### Example
+
+
+Below is an example of retrying three times to the same host
+
+@@snip [RedisRetryClient.scala](../scala/RedisRetryClient.scala) { #example }
+
+
+### HTTP Example
+
+Below is a simple Http Request example.  When using an Http Call,
+a implicit workerRef is required when using the Callback interface.
+An implicit ClientCodecProvider is required.
+
+@@snip [HttpClient.scala](../scala/HttpClient.scala) { #example }
+
+
+### Memcached Example
+
+Below is a simple memcached get example.  It can be tested locally
+by starting up a memcached server and setting a value for the key of 1.
+The equivalent command would be "get 1" from a telnet session on 11211.  
+When using the memcached client an implicit workerRef is required when using the Callback
+interface.  An implicit ClientCodeProvider is required.
+
+@@snip [MemcacheClient.scala](../scala/MemcacheClient.scala) { #example }
+
+
+### Redis Example
+
+Below is a simple redis get example.  It can be tested locally
+by starting up a redis server and setting a value for the key of 1.
+When using the Redis client an implicit workerRef is required when using the Callback
+interface.  An implicit ClientCodeProvider is required.
+
+
+@@snip [RedisClientExample.scala](../scala/RedisClientExample.scala) { #example }
+
+
+### Retry Policy
+
+In the event of a Connection Failure, the service client uses a 
+[RetryPolicy](https://tumblr.github.io/colossus/api/index.html#colossus.core.RetryPolicy)
+for re-establishing connections.  Each service client takes in a RetryPolicy on
+creation.  The client defaults to an exponential backoff starting at 50 
+milliseconds and with a maximum of 5 seconds.  The policy type can be either 
+[NoRetry](https://tumblr.github.io/colossus/api/index.html#colossus.core.NoRetry) 
+or [BackoffPolicy](https://tumblr.github.io/colossus/api/index.html#colossus.core.BackoffPolicy).
+
+A Backoff retry policy contains a multiplier which can be:
+ 
+ * Constant
+ * Linear
+ * Exponential
```

Comment:
I cut the bit about "Using clients generically" (http://tumblr.github.io/colossus/docs/clients/) since it didn't work and not sure how to make it work. Guessing high kinded type needed?